### PR TITLE
feat: introduce workflow lockfile

### PR DESCRIPTION
**Pull request by chase-crumbaugh**
Will be used for speakeasyVersion tracking and source revision tracking

### CODE REVIEW

File: workflow/lockfile.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+	SpeakeasyVersion string                `yaml:"speakeasyVersion"`
+	Sources          map[string]SourceLock `yaml:"sources"`
+	Targets          map[string]TargetLock `yaml:"targets"`
+
+	Workflow Workflow `yaml:"workflow"`
+}
+
+type SourceLock struct {
+	SourceRevisionDigest string   `yaml:"sourceRevisionDigest,omitempty"`
+	SourceNamespaceName  string   `yaml:"sourceNamespaceName,omitempty"`
```
Comment (ryan-timothy-albert) [inline workflow/lockfile.go:13]: nit: why not just sourceNamespace?

File: workflow/workflow.go [hunk 2]
```diff
@@ -21,7 +21,6 @@ const (
 type Workflow struct {
 	Version          string            `yaml:"workflowVersion"`
 	SpeakeasyVersion Version           `yaml:"speakeasyVersion,omitempty"`
-	VersionLocked    bool              `yaml:"versionLocked,omitempty"`
```
Comment (ryan-timothy-albert) [inline workflow/workflow.go:24]: What was this being used for initially? Was it something you added?

**Review by ryan-timothy-albert — APPROVED** [review]

File: workflow/lockfile.go (hunk 1 above)
Comment (chase-crumbaugh) [reply workflow/lockfile.go:25]: Oh ya that's better

File: workflow/workflow.go (hunk 2 above)
Comment (chase-crumbaugh) [reply workflow/workflow.go:24]: Yes it was, now no longer needed

**Pull request merged**